typescript/packages/plugins/uniswap/src/index.ts

```diff
@@ -0,0 +1,24 @@
+import type { Chain, EVMWalletClient, Plugin } from "@goat-sdk/core";
+import { arbitrum, avalanche, base, celo, mainnet, optimism, polygon, zora } from "viem/chains";
+import { getTools } from "./tools";
+
+export type UniswapOptions = {
+    apiKey: string;
+    baseUrl: string;
+};
+
+const SUPPORTED_CHAINS = [mainnet, polygon, avalanche, base, optimism, zora, arbitrum, celo];
+
+export function uniswap({ apiKey, baseUrl }: UniswapOptions): Plugin<EVMWalletClient> {
+    return {
+        name: "Uniswap",
+        supportsChain: (chain: Chain) => chain.type === "evm" && SUPPORTED_CHAINS.some((c) => c.id === chain.id),
+        supportsSmartWallets: () => false,
```

Comment:
Based on the code it does seem like this plugin would support smart wallets! :)

Reply:
Yep, good callout!